tests/Integration/Momento.Sdk.Tests/CacheDataTest.cs

```diff
@@ -11,6 +14,160 @@ public CacheDataTest(CacheClientFixture fixture) : base(fixture)
+        response = await client.KeysExistAsync(cacheName, oneNullKey);
+        Assert.True(response is CacheKeysExistResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((CacheKeysExistResponse.Error)response).ErrorCode);
+    }
+
+    [Fact]
+    public async Task KeysExistAsync_ByteArray()
+    {
+        byte[] key1 = Encoding.UTF8.GetBytes("foo");
+        byte[] key2 = Encoding.UTF8.GetBytes("bar");
+        byte[] key3 = Encoding.UTF8.GetBytes("baz");
+
+        await client.SetAsync(cacheName, key1, key1);
+        await client.SetAsync(cacheName, key3, key3);
+
+        var keys = new byte[][] { key1, key2, key3 };
+        CacheKeysExistResponse response = await client.KeysExistAsync(cacheName, keys);
+        Assert.True(response is CacheKeysExistResponse.Success, $"Unexpected response: {response}");
+        var goodResponse = (CacheKeysExistResponse.Success)response;
+        Assert.Equal(new List<bool> { true, false, true }, goodResponse.ExistsEnumerable.ToList());
+
+        var keysInADifferentOrder = new byte[][] { key2, key3, key1 };
+        response = await client.KeysExistAsync(cacheName, keysInADifferentOrder);
+        Assert.True(response is CacheKeysExistResponse.Success, $"Unexpected response: {response}");
+        goodResponse = (CacheKeysExistResponse.Success)response;
+        Assert.Equal(new List<bool> { false, true, true }, goodResponse.ExistsEnumerable.ToList());
+    }
+
+    [Fact]
+    public async Task KeysExistAsync_String()
+    {
+        var key1 = "foo";
+        var key2 = "bar";
+        var key3 = "baz";
+
+        await client.SetAsync(cacheName, key1, key1);
+        await client.SetAsync(cacheName, key3, key3);
+
+        var keys = new string[] { key1, key2, key3 };
+        CacheKeysExistResponse response = await client.KeysExistAsync(cacheName, keys);
+        Assert.True(response is CacheKeysExistResponse.Success, $"Unexpected response: {response}");
+        var goodResponse = (CacheKeysExistResponse.Success)response;
+        Assert.Equal(new List<bool> { true, false, true }, goodResponse.ExistsEnumerable.ToList());
+
+        var expectedDict = new Dictionary<string, bool>
+        {
+            [key1] = true,
+            [key2] = false,
+            [key3] = true
+        };
+        Assert.Equal(expectedDict, goodResponse.ExistsDictionary);
+
+        var keysInADifferentOrder = new string[] { key2, key3, key1 };
+        response = await client.KeysExistAsync(cacheName, keysInADifferentOrder);
+        Assert.True(response is CacheKeysExistResponse.Success, $"Unexpected response: {response}");
+        goodResponse = (CacheKeysExistResponse.Success)response;
+        Assert.Equal(new List<bool> { false, true, true }, goodResponse.ExistsEnumerable.ToList());
+        Assert.Equal(expectedDict, goodResponse.ExistsDictionary);
```

Comment:
I felt it's a bit confusing that the order of `ExistsEnumerable` is the same order as the passed keys, `{ key2, key3, key1 }`, but the order of `ExistsDictionary` is different, `{
            [key1] = true,
            [key2] = false,
            [key3] = true
        }`.
        
Is this expected? Am I missing something here?

Reply:
The basic Dictionary/Map/Associative Array classes in almost all programming languages are un-ordered.  So the equality check is just based on whether the key/value pairs all match.

Reply:
Got it.